lib/with_advisory_lock/concern.rb

```diff
@@ -40,24 +39,12 @@ def current_advisory_lock
 
       private
 
-      def impl_class(options = nil)
+      def impl_class
         adapter = WithAdvisoryLock::DatabaseAdapterSupport.new(connection)
         if adapter.postgresql?
           WithAdvisoryLock::PostgreSQL
         elsif adapter.mysql?
-          nested_lock = if options.respond_to?(:fetch) && [true,
-                                                           false].include?(options.fetch(:force_nested_lock_support,
-                                                                                         nil))
-                          options.fetch(:force_nested_lock_support)
-                        else
-                          adapter.mysql_nested_lock_support?
-                        end
-
-          if nested_lock
-            WithAdvisoryLock::MySQL
-          else
-            WithAdvisoryLock::MySQLNoNesting
-          end
+          WithAdvisoryLock::MySQL
```

Comment:
If we're dropping support for MySQL versions that don't support shared locks then there's a few more lines to change:

Remove:
https://github.com/ClosureTree/with_advisory_lock/blob/00f1cfe81c231aabf1f8c7c3c2b7be844df031c6/lib/with_advisory_lock/mysql.rb#L8